valohai_yaml/objs/workload_resources.py

```diff
@@ -0,0 +1,82 @@
+from collections import OrderedDict
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.types import SerializedDict
+
+
+class WorkloadResources(Item):
+    """
+    Represents a workload resource definition.
+
+    Resources include: cpu, memory, devices.
+    Both the resources and their properties are optional.
+    """
+
+    def __init__(
+        self,
+        *args: OrderedDict,
+    ) -> None:
+        self.cpu: ResourceCPU | None = None
+        self.memory: ResourceMemory | None = None
+        self.devices: ResourceDevices | None = None
+        self._parse_args(args[0])
+
+    def __repr__(self) -> str:
+        """Resources contents."""
+        return f'WorkflowResources("cpu": {self.cpu}, "memory": {self.memory}, "devices": {self.devices})'
+
+    def _parse_args(self, resources: dict) -> None:
+        if not resources:
+            return
+
+        if cpu := resources.get("cpu"):
+            self.cpu = ResourceCPU(cpu)
+        if memory := resources.get("memory"):
+            self.memory = ResourceMemory(memory)
+        if devices := resources.get("devices"):
+            self.devices = ResourceDevices(devices)
+
+
+class ResourceCPU(Item):
+    """CPU configuration."""
+
+    # TODO after removing support for Python 3.8 set cpu_resource type: OrderedDict[str, int]
+    def __init__(self, cpu_resource) -> None:  # noqa: ANN001
```

Comment:
for the typing, probably `Dict[str, Any]` should be right call that we have aliased as `SerializedDict` e.g. https://github.com/valohai/valohai-yaml/blob/master/valohai_yaml/objs/endpoint.py#L36

but as requested previously, `ResourceCPU.__init__` should take simply `min: int, max: int` and a `parse` override should be taking a dictionary for the parsing of serialized objects

Reply:
That took care of the typing! 👍